app/models/dialog_field_association_validator.rb

```diff
@@ -0,0 +1,25 @@
+class DialogFieldAssociationValidator
+  def circular_references(associations)
+    return false if associations.empty?
+    association_path_list = initial_paths(associations)
+    association_path_list.each do |path|
+      fieldname_being_triggered = path.last
+      next if associations[fieldname_being_triggered].blank?
+      return walk_value_path(fieldname_being_triggered, associations, path) if walk_value_path(fieldname_being_triggered, associations, path)
+    end
+    false
+  end
+
+  def initial_paths(associations)
```

Comment:
`private` should be right above this, I think, since we have no intention of calling these methods from outside this class.